maxim/apis/maxim_apis.py

```
@@ -110,47 +117,187 @@ def __init__(self, base_url: str, api_key: str):
         self.connection_pool = ConnectionPool()
         self.base_url = base_url
         self.api_key = api_key
+        self.max_retries = 3
+        self.base_delay = 1.0
 
-    def __make_network_call(
+    def __make_network_call_with_retry(
```

Ruff (0.11.9)

`__make_network_call_with_retry` is too complex (18 > 10)

(C901)